apps/server/src/modules/task/service/submission.service.ts

```diff
@@ -26,4 +31,104 @@ export class SubmissionService {
 
 		await this.submissionRepo.delete(submission);
 	}
+
+	async deleteSubmissionsByUserId(userId: EntityId): Promise<DomainOperation> {
```

Comment:
I know naming is hard and also subjective, but I think it would be better to have some more meaningful name here like `deleteSubmissionsOwnedByUser` or something that would explain a little bit what's the relation between the user and the submissions you want to delete here.